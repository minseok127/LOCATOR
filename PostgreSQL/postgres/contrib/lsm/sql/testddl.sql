--
-- Test create, alter, and drop a table
--

\c lsmtest

CREATE FOREIGN TABLE testddl(id SERIAL, name VARCHAR(20), price NUMERIC(10,2), inventory INT, stime TIMESTAMP, flag BOOLEAN) SERVER lsm_server;

INSERT INTO testddl VALUES (DEFAULT, 'Name1', 10.00, 50, current_timestamp(2), false);
INSERT INTO testddl VALUES (DEFAULT, 'Name2', 20.00, 50, current_timestamp(2), false);
INSERT INTO testddl VALUES (DEFAULT, 'Name3', 30.00, 50, current_timestamp(2), false);

\d testddl

SELECT id,name,price,inventory,flag FROM testddl;

ALTER FOREIGN TABLE testddl RENAME TO item;

\d testddl

\d item

ALTER FOREIGN TABLE item RENAME COLUMN stime TO stamp;

\d item

ALTER FOREIGN TABLE item ALTER COLUMN name TYPE TEXT;

\d item

SELECT id,name,price,inventory,flag FROM item;

ALTER FOREIGN TABLE item DROP COLUMN flag;

\d item

DROP FOREIGN TABLE item;
