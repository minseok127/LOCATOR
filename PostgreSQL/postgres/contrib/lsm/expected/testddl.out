--
-- Test create, alter, and drop a table
--
\c lsmtest
CREATE FOREIGN TABLE testddl(id SERIAL, name VARCHAR(20), price NUMERIC(10,2), inventory INT, stime TIMESTAMP, flag BOOLEAN) SERVER lsm_server;
INSERT INTO testddl VALUES (DEFAULT, 'Name1', 10.00, 50, current_timestamp(2), false);
INSERT INTO testddl VALUES (DEFAULT, 'Name2', 20.00, 50, current_timestamp(2), false);
INSERT INTO testddl VALUES (DEFAULT, 'Name3', 30.00, 50, current_timestamp(2), false);
\d testddl
                                           Foreign table "public.testddl"
  Column   |            Type             | Collation | Nullable |               Default               | FDW options 
-----------+-----------------------------+-----------+----------+-------------------------------------+-------------
 id        | integer                     |           | not null | nextval('testddl_id_seq'::regclass) | 
 name      | character varying(20)       |           |          |                                     | 
 price     | numeric(10,2)               |           |          |                                     | 
 inventory | integer                     |           |          |                                     | 
 stime     | timestamp without time zone |           |          |                                     | 
 flag      | boolean                     |           |          |                                     | 
Server: lsm_server

SELECT id,name,price,inventory,flag FROM testddl;
 id | name  | price | inventory | flag 
----+-------+-------+-----------+------
  1 | Name1 | 10.00 |        50 | f
  2 | Name2 | 20.00 |        50 | f
  3 | Name3 | 30.00 |        50 | f
(3 rows)

ALTER FOREIGN TABLE testddl RENAME TO item;
\d testddl
\d item
                                            Foreign table "public.item"
  Column   |            Type             | Collation | Nullable |               Default               | FDW options 
-----------+-----------------------------+-----------+----------+-------------------------------------+-------------
 id        | integer                     |           | not null | nextval('testddl_id_seq'::regclass) | 
 name      | character varying(20)       |           |          |                                     | 
 price     | numeric(10,2)               |           |          |                                     | 
 inventory | integer                     |           |          |                                     | 
 stime     | timestamp without time zone |           |          |                                     | 
 flag      | boolean                     |           |          |                                     | 
Server: lsm_server

ALTER FOREIGN TABLE item RENAME COLUMN stime TO stamp;
\d item
                                            Foreign table "public.item"
  Column   |            Type             | Collation | Nullable |               Default               | FDW options 
-----------+-----------------------------+-----------+----------+-------------------------------------+-------------
 id        | integer                     |           | not null | nextval('testddl_id_seq'::regclass) | 
 name      | character varying(20)       |           |          |                                     | 
 price     | numeric(10,2)               |           |          |                                     | 
 inventory | integer                     |           |          |                                     | 
 stamp     | timestamp without time zone |           |          |                                     | 
 flag      | boolean                     |           |          |                                     | 
Server: lsm_server

ALTER FOREIGN TABLE item ALTER COLUMN name TYPE TEXT;
\d item
                                            Foreign table "public.item"
  Column   |            Type             | Collation | Nullable |               Default               | FDW options 
-----------+-----------------------------+-----------+----------+-------------------------------------+-------------
 id        | integer                     |           | not null | nextval('testddl_id_seq'::regclass) | 
 name      | text                        |           |          |                                     | 
 price     | numeric(10,2)               |           |          |                                     | 
 inventory | integer                     |           |          |                                     | 
 stamp     | timestamp without time zone |           |          |                                     | 
 flag      | boolean                     |           |          |                                     | 
Server: lsm_server

SELECT id,name,price,inventory,flag FROM item;
 id | name  | price | inventory | flag 
----+-------+-------+-----------+------
  1 | Name1 | 10.00 |        50 | f
  2 | Name2 | 20.00 |        50 | f
  3 | Name3 | 30.00 |        50 | f
(3 rows)

ALTER FOREIGN TABLE item DROP COLUMN flag;
\d item
                                            Foreign table "public.item"
  Column   |            Type             | Collation | Nullable |               Default               | FDW options 
-----------+-----------------------------+-----------+----------+-------------------------------------+-------------
 id        | integer                     |           | not null | nextval('testddl_id_seq'::regclass) | 
 name      | text                        |           |          |                                     | 
 price     | numeric(10,2)               |           |          |                                     | 
 inventory | integer                     |           |          |                                     | 
 stamp     | timestamp without time zone |           |          |                                     | 
Server: lsm_server

DROP FOREIGN TABLE item;
